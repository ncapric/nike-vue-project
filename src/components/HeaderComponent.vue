<template>
  <div class="header">
    <img src="../assets/icons/site-logo.svg" alt="Logo" class="logo" />
    <div class="right-side">
      <div v-for="social in socialMedia" :key="social.altText">
        <a :href="social.pageLink">
          <img :src="social.imgSrc" :alt="social.altText" />
        </a>
      </div>
      <h2>Welcome: {{ randomUserInitials }}</h2>
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import { socialMedia } from '@/constants/social-media-links';
import { useStore } from 'vuex';

const store = useStore();

const randomUserInitials = computed(() => store.getters.getRandomUserInitials);

</script>

<style scoped>
.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #028c6a;
  width: 90vw;
}
.right-side {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

img {
  /* width: auto; */
  height: 20px;
  margin: 0 10px;
}

.logo {
  height: 70px;
}
</style>
