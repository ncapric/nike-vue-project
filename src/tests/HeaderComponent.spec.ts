import { useStore } from 'vuex';

import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import HeaderComponent from '@/components/HeaderComponent.vue';



describe('Header Tests', async () => {

    const store = useStore();

    it("renders the header", () => {
        const wrapper = mount(HeaderComponent, {
            global: {
                plugins: [store]
            }
        })
 
        const div = wrapper.find('div.header');
        expect(div.exists()).toBe(true);
    })

    it('displays the correct social media links', () => {
        const wrapper = mount(HeaderComponent);
        const socialLinks = wrapper.findAll('.right-side a');

        expect(socialLinks).toHaveLength(3);

        expect(socialLinks[0].attributes('href')).toBe('https://www.facebook.com/');
        expect(socialLinks[0].find('img').attributes('alt')).toBe('Facebook');
        expect(socialLinks[0].find('img').attributes('src')).toBe('/assets/icons/Facebook.png');
    });

    it('displays the user initials', () => {
        const wrapper = mount(HeaderComponent, {
            global: {
                plugins: [store]
            }
        })

        expect(wrapper.html()).toMatchSnapshot();
    })

})