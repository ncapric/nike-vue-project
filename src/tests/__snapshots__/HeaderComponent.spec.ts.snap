// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Tests > displays the user initials 1`] = `
"<div data-v-727bebfc=\\"\\" class=\\"header\\"><img data-v-727bebfc=\\"\\" src=\\"/src/assets/icons/site-logo.svg\\" alt=\\"Logo\\" class=\\"logo\\">
  <div data-v-727bebfc=\\"\\" class=\\"right-side\\">
    <div data-v-727bebfc=\\"\\"><a data-v-727bebfc=\\"\\" href=\\"https://www.facebook.com/\\"><img data-v-727bebfc=\\"\\" src=\\"/assets/icons/Facebook.png\\" alt=\\"Facebook\\"></a></div>
    <div data-v-727bebfc=\\"\\"><a data-v-727bebfc=\\"\\" href=\\"https://www.youtube.com/\\"><img data-v-727bebfc=\\"\\" src=\\"/assets/icons/Youtube.png\\" alt=\\"Youtube\\"></a></div>
    <div data-v-727bebfc=\\"\\"><a data-v-727bebfc=\\"\\" href=\\"https://twitter.com/\\"><img data-v-727bebfc=\\"\\" src=\\"/assets/icons/Twitter.svg\\" alt=\\"Twitter\\"></a></div>
    <h2 data-v-727bebfc=\\"\\">Welcome: </h2>
  </div>
</div>"
`;
